rust/pattern/constraint/type_/owns.rs

```diff
@@ -89,113 +79,202 @@ impl Validatable for OwnsConstraint {
 
 impl From<&str> for OwnsConstraint {
     fn from(attribute_type: &str) -> Self {
-        OwnsConstraint::from((attribute_type, IsKeyAttribute::No))
+        OwnsConstraint::from(Label::from(attribute_type))
     }
 }
```

Comment:
A little more systematic in how the `From<T>` are implemented for the constraint:

- `&str`/`String` are transformed into `Label`;
- `Label`, `Type`, and `UnboundVariable` are all transformed into `TypeVariable`;
- single `Annotation` — into an array of `Annotation`s.

Reply:
These could potentially be generated by a clever macro based on the rules like the above, to simplify the cognitive load.